src/cluster/src/check.rs

```diff
@@ -40,6 +51,7 @@ pub enum CheckError {
     /// Failed to parse kubernetes cluster server URL
     #[error("Failed to parse server url from Kubernetes context")]
     BadKubernetesServerUrl {
```

Comment:
So there are couple of problem with this:
1) I think this is becoming unwieldy.  Every time we create new check, this needs to be changed. 
2) Result of check should not be Error.  It is expected Condition.  Rather You could create individual message for result of check

Reply:
I agree, but these errors are being used in installation workflow to fix the specific condition. So if we do not use these errors, then we will have to change the logic in the installation too for self fixing pre install issues. I am re-using the same logic in both pre flight and pre install checks.
cc @nicholastmosher

Reply:
@sehz the `CheckError` type represents the various kinds of failed checks which may occur. It is not intended to be used strictly as a "bubble-up" type of error, but one which we examine to see if any course of action may be taken in order to fix it. Just because we call this type "Error" does not mean we have to treat it as unrecoverable. The Rust Book itself describes this as one of the benefits of the `Result` type https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html

Reply:
Yeah these are more like java exceptions which can be handled and I like the way we can embed message here. @sehz any inputs here?

Reply:
Error generally means something unexpected.  It can be recoverable or unrecoverable but it is not happy path.  For example, I am reading a file, I am expecting file to be there. If file is not there, it returns error.

In this case, we are doing status checking.  Expectation is that some part of system may or not be in right state.  That is expected behavior.  For example, network may be down.  That is expected condition.   What is not expected is that network card doesn't exist or we issue with the network driver.

Problem with returning check as Error is that we can't distinguish between expected behavior vs something unexpected.   So instead returning String,  we should return as Enum.   Something like:  

pub enum StatusCheck {
       Working,  // or could be Normal
       NotWorkingNoRemediation(Fault),   // doesn't work and there is nothing we could do about it
       NotWorking(Remediation),     // doesn't work but there is a way to fix it
}


It could be more sophisticated later like indicating Software/Hardware/Version/ etc.

Reply:
> Problem with returning check as Error is that we can't distinguish between expected behavior vs something unexpected. So instead returning String, we should return as Enum.

I don't really understand what you mean here. The `CheckError` type _is_ an enum, and in the present codebase we are distinguishing between recoverable and unrecoverable errors, [here](https://github.com/infinyon/fluvio/blob/master/src/cluster/src/install.rs#L626-L655).

I feel like adding this `StatusCheck` type would only add more indirection to the code and make it more difficult to wrangle everything into the right places, but I suppose that's just my opinion so ¯\_(ツ)_/¯

Reply:
CheckError is enum but it represents Error.   But Status Check should not be  Error to differentiate from other error

Reply:
@sehz I have added the changes for StatusCheck. Do these changes look good to you?